### DISCUSSION TITLE
Define dependencies between projects for ready checks

### DISCUSSION THREAD

**Discussion by christiannagel** [microsoft/aspire#2873] [Q&A] [answered]
Having dependencies between projects, such as with a database migration https://github.com/dotnet/aspire-samples/tree/main/samples/DatabaseMigrations, is it possible to define dependencies, such as the API service is only available for requests when the migrations project finished?

**Comment by davidfowl** [comment, marked as answer]
No there’s no support for dependencies

**Comment by christiannagel** [comment]
@davidfowl this could be interesting e.g. with an EF Core migrations project that needs to run before the main container. This would be solved with *init containers* support #2866 with Azure Container Apps where the init container runs in the same app. I don't know what the equivalent would be with Kubernetes.

**Reply by christiannagel** [reply]
Update: `WaitFor`, `WaitForCompletion` https://twitter.com/davidfowl/status/1787696560559792458

https://github.com/davidfowl/WaitForDependenciesAspire/blob/4c37c94f8a9d9ba349883f27737e5bead6cce803/WaitForDependencies.Aspire.Hosting/WaitForDependenciesExtensions.cs
